app/org/sagebionetworks/bridge/services/AccountWorkflowService.java

```diff
@@ -0,0 +1,233 @@
+            return userId;
+        }
+    }
+    
+    private StudyService studyService;
+    
+    private SendMailService sendMailService;
+    
+    private AccountDao accountDao;
+    
+    private CacheProvider cacheProvider;
+    
+    @Autowired
+    final void setStudyService(StudyService studyService) {
+        this.studyService = studyService;
+    }
+    
+    @Autowired
+    final void setSendMailService(SendMailService sendMailService) {
+        this.sendMailService = sendMailService;
+    }
+    
+    @Autowired
+    final void setAccountDao(AccountDao accountDao) {
+        this.accountDao = accountDao;
+    }
+    
+    @Autowired
+    final void setCacheProvider(CacheProvider cacheProvider) {
+        this.cacheProvider = cacheProvider;
+    }
+    
+    /**
+     * Send email verification token as part of creating an account that requires an email address be
+     * verified. We assume that an account has been created and that email verification should be sent
+     * (neither is verified in this method).
+     */
+    public void sendEmailVerificationToken(Study study, String userId, String email) {
+        checkNotNull(study);
+        checkArgument(isNotBlank(userId));
+        checkArgument(isNotBlank(email));
+
+        String sptoken = createTimeLimitedToken();
+        
+        saveVerification(sptoken, new VerificationData(study.getIdentifier(), userId));
+        
+        String studyId = BridgeUtils.encodeURIComponent(study.getIdentifier());
+        String url = String.format(VERIFY_EMAIL_URL, BASE_URL, studyId, sptoken);
+        
+        BasicEmailProvider provider = new BasicEmailProvider.Builder()
+            .withStudy(study)
+            .withEmailTemplate(study.getVerifyEmailTemplate())
+            .withRecipientEmail(email)
+            .withToken(URL_TOKEN, url).build();
+        sendMailService.sendEmail(provider);         
+    }
+    
+    /**
+     * Send another email verification token. This creates and sends a new verification token 
+     * starting with the user's email address.
+     */
+    public void resendEmailVerificationToken(StudyIdentifier studyIdentifier, Email email) {
+        checkNotNull(studyIdentifier);
+        checkNotNull(email);
+        
+        Study study = studyService.getStudy(studyIdentifier);
+        Account account = accountDao.getAccountWithEmail(study, email.getEmail());
+        if (account != null) {
+            sendEmailVerificationToken(study, account.getId(), email.getEmail());
+        }
+    }
+    
+    /**
+     * Using the verification token that was sent to the user, verify the email address. 
+     * If the token is invalid, it fails quietly. If the token exists but the account 
+     * does not, it throws an exception (this would be unexpected).
+     */
+    public void verifyEmail(EmailVerification verification) {
+        checkNotNull(verification);
+
+        VerificationData data = restoreVerification(verification.getSptoken());
```

Comment:
This is not the current behavior but I agree with you that it's best to return a 400 here. Changed it.